docs/deployment.md

```diff
@@ -1,66 +1,182 @@
-There are two deployment artifacts:
+## GCP runtime environment
+
+BigFlow GCP runtime environment consists of two services:
+
+1. Google [Cloud Composer](#cloud-composer),
+2. [Docker Registry](#docker-registry).
+
+Typically, for one software project, teams use one or more
+GCP projects (for dev, test, and prod environments)
+and one long-running Composer instance per each GCP project.
+
+Docker images are heavy files, so pushing them only once to GCP greatly
+reduces subsequent deployment time (it's safe because images are immutable).
+That's why we recommend using a single Docker Registry instance of,
+shared by all environments. 
+  
+There are two [deployment artifacts](project_setup_and_build.md#deployment-artifacts):
 
 1. Airflow DAG files with workflows definitions,
-1. Docker image with workflows computation code.
+1. a Docker image with workflows computation code.
 
-## GCP runtime environment
+During deployment, BigFlow uploads your DAG files to Composer's [DAGs folder](#composers-dags-folder)
+and pushes your Docker image to Docker Registry.
+
+Read more about deployment artifacts in [Project setup and build](project_setup_and_build.md).
+
+## Cloud Composer
+
+Shortly speaking, a Cloud Composer is Airflow-as-a-Service.
+
+Unfortunately for Python users, Composer's architecture is flawed
+because Python libraries required by DAGs have to be
+[installed manually](https://cloud.google.com/composer/docs/how-to/using/installing-python-dependencies) on Composer.
+To make it worse, installing dependencies forces a Composer instance to restart.
+It not only takes time but often fails. In the worst scenario, you need to spawn a new Composer instance.
+ 
+BigFlow fixes these problems by using Docker. Each of your [jobs](workflow-and-job.md)
+is executed in a stable and isolated runtime environment &mdash; a Docker container.
+
+On GCP you execute Docker images on Kubernetes.
+BigFlow leverages the fact that each Composer instance
+stands on its own ([GKE](https://cloud.google.com/kubernetes-engine)) cluster.
+This cluster is reused by BigFlow.
+
+### Composer's service account
 
-**//TODO** 
+Before you start you will need a [GCP project](https://cloud.google.com/resource-manager/docs/creating-managing-projects)
+and a [Service account](https://cloud.google.com/iam/docs/service-accounts). 
 
-### Cloud Composer
+That's important. All permissions required by a Composer itself and by your jobs have to be granted to this account.
 
-**//TODO** 
+We recommend using a [default service account](https://cloud.google.com/compute/docs/access/service-accounts#default_service_account)
+as a Composer's account.
+This account is created automatically for each GCP project. It has the following email:
+
+```
+PROJECT_NUMBER-compute@developer.gserviceaccount.com
+```
+
+### Setting up a Composer Instance
+
+[Create](https://cloud.google.com/composer/docs/quickstart)
+a new Composer instance. Set only these properties (the others leave blank or default):
```

Comment:
Jeszcze wersja pythona jaka trzeba wybrac to 3

Reply:
ale sprzawdziłem że jak nie wypełnisz, to defaultuje się na 3, więc może nie warto o tym pisać

Reply:
ok